storage/integration_test.go

```diff
@@ -1236,6 +1237,42 @@ func testObjectsIterateAllSelectedAttrs(t *testing.T, bkt *BucketHandle, objects
 	}
 }
 
+func testObjectsIterateWithProjection(t *testing.T, bkt *BucketHandle) {
+	projections := map[Projection]bool{
+		ProjectionDefault: true,
+		ProjectionFull:    true,
+		ProjectionNoACL:   false,
+	}
+
+	for projection, expectACL := range projections {
+		query := &Query{Projection: projection}
+		it := bkt.Objects(context.Background(), query)
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			t.Fatalf("no objects")
+		}
+		if err != nil {
+			t.Fatalf("iterator.Next: %v", err)
+		}
+
+		if expectACL {
+			if attrs.Owner == "" {
+				t.Errorf("projection: %q, empty Owner", projection)
```

Comment:
nit: errors should be in the format of "got something, want something else": https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures